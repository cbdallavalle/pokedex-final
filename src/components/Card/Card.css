.Card {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 300px;
  height: 70px;
  border: 1px solid #a8a8a8;
  margin: 5px;
  box-shadow: 5px 5px 5px grey;
}

.pokemon-info-cont {
  width: 100%;
  height: 340px;
  overflow-y: scroll;
}

#displayPokemon {
  height: 400px;
}