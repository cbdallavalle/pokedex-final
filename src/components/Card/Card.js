import React, { Component } from 'react';
import { PokemonInfo } from '../PokemonInfo/PokemonInfo';

export class Card extends Component {
  constructor() {
    super();
    this.state = {
      clicked: false
    }
  }

  handleClick = () => {
    this.setState({ clicked: !this.state.clicked })
  }

  pokemonInfo = () => {
    return this.props.pokemonToDisplay && this.state.clicked === true 
      && this.props.pokemonToDisplay.map( (pokemon, index) => {
      return (
        <PokemonInfo key={ index } {...pokemon} />
      )
    })
  }

  render() {
    return (
      <article 
        className="Card"
        onClick={ () => {
          this.handleClick();
          this.props.handleClick(this.props.type)
        } }
      >
        <h1>{ this.props.type }</h1>
        { this.pokemonInfo() }
      </article>
    )
  }
} 