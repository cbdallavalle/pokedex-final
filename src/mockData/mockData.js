export const mockData = {
  mockPokeTypes: {
    normal: ['16', '17', '18'],
    fighting: ['56', '57', '58'],
    flying: ['21', '22', '23']
  },
  mockRawPokeTypes: [
    {id: "1", name: "normal", pokemon: ['16', '17', '18'] },
    {id: "2", name: "fighting", pokemon: ['56', '57', '58'] },
    {id: "3", name: "flying", pokemon: ['21', '22', '23'] }
  ],
  expectedStoreTypes: {
    type: 'UPLOAD_TYPES',
    pokeTypes: {
    normal: ['16', '17', '18'],
    fighting: ['56', '57', '58'],
    flying: ['21', '22', '23']
    }
  },
  mockPokemonToDisplay: [
    {weight: 18, name: "pidgey", icon: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png", type: "normal"},
    {weight: 300, name: "pidgeotto", icon: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png", type: "normal"}
  ],
  mockRawPokemonData: {
    id: "16",
    name: "pidgey",
    sprites: {
      back_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/16.png",
      front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png"
    },
    type: "1",
    weight: 18
  },
  mockCleanPokemonData: [
    {
      weight: 18,
      name: "pidgey",
      icon: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png",
      type: "flying"
    }
  ]
}