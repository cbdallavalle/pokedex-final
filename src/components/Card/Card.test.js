import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';
import { mockData } from '../../mockData/mockData';

describe("Card", () => {
  let wrapper;
  const mockHandleClick = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Card 
        type={ 'electric' }
        handleClick={ mockHandleClick }
        pokemonToDisplay={ mockData.mockPokemonToDisplay }
      />
    )
  })

  it('should match the snapshot when clicked is false and pokemonToDisplay does not exist', () => {
    const wrapper = shallow(
      <Card 
        type={ 'electric' }
        handleClick={ mockHandleClick }
      />
    )

    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when clicked is true and pokemonToDisplay exists', () => {
    wrapper.setState({ clicked: true })
    expect(wrapper).toMatchSnapshot();

  })

  it('should match the snapshot when clicked is fasle but pokemonToDisplay exists', () => {
    expect(wrapper).toMatchSnapshot();
  })

  //handleClick
  it('handleClick should update the state to be true if clicked is false', () => {
    expect(wrapper.state().clicked).toEqual(false);
    wrapper.instance().handleClick();
    expect(wrapper.state().clicked).toEqual(true);
  })

  it('handleClick should update the state to be false if clicked is true', () => {
    wrapper.setState({clicked: true})
    expect(wrapper.state().clicked).toEqual(true);
    wrapper.instance().handleClick();
    expect(wrapper.state().clicked).toEqual(false);
  })

  //props handleClick
  it('should call props handleClick when article is clicked', () => {

    wrapper.find('article').simulate('click');
    expect(mockHandleClick).toHaveBeenCalledWith('electric')
  })
})