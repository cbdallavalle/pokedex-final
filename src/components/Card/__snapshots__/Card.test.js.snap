// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot when clicked is false and pokemonToDisplay does not exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    handleClick={[MockFunction]}
    type="electric"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          electric
        </h1>,
        <div
          className="pokemon-info-cont"
        />,
      ],
      "className": "Card",
      "id": "",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "electric",
        },
        "ref": null,
        "rendered": "electric",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "pokemon-info-cont",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            electric
          </h1>,
          <div
            className="pokemon-info-cont"
          />,
        ],
        "className": "Card",
        "id": "",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "electric",
          },
          "ref": null,
          "rendered": "electric",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "pokemon-info-cont",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot when clicked is fasle but pokemonToDisplay exists 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    handleClick={[MockFunction]}
    pokemonToDisplay={
      Array [
        Object {
          "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png",
          "name": "pidgey",
          "type": "normal",
          "weight": 18,
        },
        Object {
          "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png",
          "name": "pidgeotto",
          "type": "normal",
          "weight": 300,
        },
      ]
    }
    type="electric"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          electric
        </h1>,
        <div
          className="pokemon-info-cont"
        />,
      ],
      "className": "Card",
      "id": "",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "electric",
        },
        "ref": null,
        "rendered": "electric",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "pokemon-info-cont",
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            electric
          </h1>,
          <div
            className="pokemon-info-cont"
          />,
        ],
        "className": "Card",
        "id": "",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "electric",
          },
          "ref": null,
          "rendered": "electric",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "pokemon-info-cont",
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot when clicked is true and pokemonToDisplay exists 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    handleClick={[MockFunction]}
    pokemonToDisplay={
      Array [
        Object {
          "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png",
          "name": "pidgey",
          "type": "normal",
          "weight": 18,
        },
        Object {
          "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png",
          "name": "pidgeotto",
          "type": "normal",
          "weight": 300,
        },
      ]
    }
    type="electric"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          electric
        </h1>,
        <div
          className="pokemon-info-cont"
        >
          <Unknown
            icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png"
            name="pidgey"
            type="normal"
            weight={18}
          />
          <Unknown
            icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png"
            name="pidgeotto"
            type="normal"
            weight={300}
          />
        </div>,
      ],
      "className": "Card",
      "id": "displayPokemon",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "electric",
        },
        "ref": null,
        "rendered": "electric",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png"
              name="pidgey"
              type="normal"
              weight={18}
            />,
            <Unknown
              icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png"
              name="pidgeotto"
              type="normal"
              weight={300}
            />,
          ],
          "className": "pokemon-info-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png",
              "name": "pidgey",
              "type": "normal",
              "weight": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png",
              "name": "pidgeotto",
              "type": "normal",
              "weight": 300,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            electric
          </h1>,
          <div
            className="pokemon-info-cont"
          >
            <Unknown
              icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png"
              name="pidgey"
              type="normal"
              weight={18}
            />
            <Unknown
              icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png"
              name="pidgeotto"
              type="normal"
              weight={300}
            />
          </div>,
        ],
        "className": "Card",
        "id": "displayPokemon",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "electric",
          },
          "ref": null,
          "rendered": "electric",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png"
                name="pidgey"
                type="normal"
                weight={18}
              />,
              <Unknown
                icon="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png"
                name="pidgeotto"
                type="normal"
                weight={300}
              />,
            ],
            "className": "pokemon-info-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png",
                "name": "pidgey",
                "type": "normal",
                "weight": 18,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "icon": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png",
                "name": "pidgeotto",
                "type": "normal",
                "weight": 300,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
